pkg/kubelet/kuberuntime/kuberuntime_termination_order.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+
+	"k8s.io/kubernetes/pkg/kubelet/types"
+)
+
+// terminationOrdering is used to enforce a termination ordering for sidecar containers.  It sets up
```

Comment:
nit:
Do we have to call this restartable init container instead of the sidecar containers to be consistent with other code?

Reply:
Yeah, lets do it in a follow up so we don't lose the approval.